cmake/compiler_settings.cmake

```diff
@@ -11,6 +11,13 @@ if (CMAKE_C_COMPILER_ID MATCHES Clang)
   endif ()
 endif ()
 
+if (OE_SGX AND CMAKE_C_COMPILER_ID MATCHES GNU)
+  message(
+    FATAL_ERROR
```

Comment:
Let's allow GCC compilation if the EXPERIMENTAL flag is also specified.

@mikbras @bodzhang @vtikoo 
Do you build oe from source? If so, what compiler do you use?

Reply:
Now the PR is changed to give warnings instead of raising errors.

Reply:
We are not specifying any compiler for the openenclave build, and CMake picks up Clang 7.1.0.

Reply:
Thanks for the information!